src/visualize/callbacks/filter_callbacks.py

```
@@ -0,0 +1,784 @@
+                    ]
+
+        # 4. Cross Service Filter
+        if (
+            cross_service
+            and cross_service != "all"
+            and "clone_id" in df_filtered.columns
+        ):
+            try:
+                selected_clone_id = int(str(cross_service))
+                df_filtered = df_filtered[df_filtered["clone_id"] == selected_clone_id]
+            except:
```

Ruff (0.15.5)

[error] Do not use bare `except`

(E722)